homeassistant/components/deconz/config_flow.py

```diff
@@ -254,9 +254,14 @@ async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResu
         self.bridge_id = normalize_bridge_id(discovery_info.config[CONF_SERIAL])
         await self.async_set_unique_id(self.bridge_id)
 
-        self.host = discovery_info.config[CONF_HOST]
-        self.port = discovery_info.config[CONF_PORT]
-        self.api_key = discovery_info.config[CONF_API_KEY]
+        self.host: str = discovery_info.config[CONF_HOST]
+        self.port: int = discovery_info.config[CONF_PORT]
+        self.api_key: str = discovery_info.config[CONF_API_KEY]
+
+        for entry_id in self.hass.data.get(DOMAIN, []):
+            gateway: DeconzGateway = self.hass.data[DOMAIN][entry_id]
+            if gateway.bridgeid == self.bridge_id and gateway.available:
+                return self.async_abort(reason="already_configured")
```

Comment:
Doesn't this introduce a race where the new check will fail if the config entry is not yet setup?

Reply:
I didn't get this is a workaround for a bug in the addon